apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx

```diff
@@ -117,148 +112,11 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
       isInstantMeeting: eventData && queryIsInstantMeeting ? true : false,
       themeBasis: null,
       orgBannerUrl: team.parent?.bannerUrl ?? "",
-      teamMemberEmail: await handleGettingTeamMemberEmail(query, eventTypeId, eventData),
+      teamMemberEmail: await getTeamMemberEmailForResponseOrContactUsingUrlQuery({
+        query,
+        eventTypeId,
+        eventData,
+      }),
     },
   };
 };
-
-interface EventData {
-  id: number;
-  isInstantEvent: boolean;
-  schedulingType: SchedulingType | null;
-  metadata: Prisma.JsonValue | null;
-  length: number;
-}
-
-async function handleGettingTeamMemberEmail(
-  query: ParsedUrlQuery,
-  eventTypeId: number,
-  eventData: EventData
-) {
-  if (
-    !query.email ||
-    typeof query.email !== "string" ||
-    eventData.schedulingType !== SchedulingType.ROUND_ROBIN
-  )
-    return null;
-
```

Comment:
All code abstracted out so that it can be shared with Test Preview code as well.